src/Microsoft.Azure.WebJobs.Host/Listeners/FunctionListener.cs

```diff
@@ -54,7 +57,54 @@ public void Dispose()
             }
             catch (Exception e)
             {
-                new FunctionListenerException(_descriptor.ShortName, e).TryRecover(_trace, _logger);
+                var flx = new FunctionListenerException(_descriptor.ShortName, e);
+                if (_allowPartialHostStartup)
+                {
+                    flx.Handled = true;
+                }
+
+                flx.TryRecover(_trace, _logger);
+
+                if (_allowPartialHostStartup)
+                {
+                    // if to here, the listener exception was handled and
+                    // we're in partial startup mode.
+                    // we spin up a background task retrying to start the listener
+                    Task taskIgnore = Task.Run(() => RetryStartWithBackoffAsync(cancellationToken), cancellationToken);
+                }
+            }
+        }
+
+        public async Task RetryStartWithBackoffAsync(CancellationToken cancellationToken)
+        {
+            int attempt = 0;
+            while (!_started)
+            {
+                await Utility.DelayWithBackoffAsync(++attempt, cancellationToken, 
+                    unit: TimeSpan.FromSeconds(1),
+                    min: TimeSpan.FromSeconds(2),
+                    max: TimeSpan.FromMinutes(2));
+
+                try
+                {
+                    string message = $"Retrying to start listener for function '{_descriptor.ShortName}' (Attempt {attempt})";
+                    _trace.Info(message);
+                    _logger?.LogInformation(message);
+
+                    await _listener.StartAsync(cancellationToken);
+                    _started = true;
+
+                    message = $"Listener successfully started for function '{_descriptor.ShortName}' after {attempt} retries.";
+                    _trace.Info(message);
+                    _logger?.LogInformation(message);
+                }
+                catch (Exception ex) when (!ex.IsFatal())
+                {
+                    // swallow and log exceptions since we're in a retry loop
+                    // at this point doing this in the background
+                    _trace.Error(ex.Message, ex);
+                    _logger?.LogError(0, ex, ex.Message);
```

Comment:
Same as above, wondering about whether this will generate lots of log noise..

Reply:
Noise for us you mean? Or for customers - I don't think it is noise for customers. They should be seeing these.